File: src/main/java/org/micoli/php/scripting/UI.java
package org.micoli.php.scripting;

import com.intellij.openapi.project.Project;
import org.micoli.php.ui.Notification;

/**
 * known as `ui` in scripting engine
 */
public final class UI {
    private final Project project;

    public UI(Project project) {
        this.project = project;
    }
    /**
     * Displays a closable popup
     *
     * @param message the message to display.
     */
    public void alert(String message) {
        Notification.getInstance(project).message(message);
    }

    /**
     * Displays a closable popup and automatically close it after a given delay
     *
     * @param message the message to display.
     * @param delayInMs the delay in milliseconds before the popup is closed.
     */
    public void alert(String message, int delayInMs) {
        Notification.getInstance(project).messageWithTimeout(message, delayInMs);
    }
}

File: src/main/kotlin/org/micoli/php/scripting/UI.kt
package org.micoli.php.scripting

import com.intellij.openapi.project.Project
import org.micoli.php.ui.Notification.Companion.getInstance

interface ScriptUI {
    fun alert(message: String)

    fun alert(message: String, delayInMs: Int)
}

/** known as `ui` in scripting engine */
class UI(private val project: Project) : ScriptUI {
    /**
     * Displays a closable popup
     *
     * @param message the message to display.
     */
    override fun alert(message: String) {
        getInstance(project).message(message)
    }

    /**
     * Displays a closable popup and automatically close it after a given delay
     *
     * @param message the message to display.
     * @param delayInMs the delay in milliseconds before the popup is closed.
     */
    override fun alert(message: String, delayInMs: Int) {
        getInstance(project).messageWithTimeout(message, delayInMs)
    }
}
